chrome/content/zotero/xpcom/translation/translate.js

```diff
@@ -2182,15 +2188,21 @@ Zotero.Translate.Export.prototype._prepareTranslation = function() {
 	this._itemGetter = new Zotero.Translate.ItemGetter();
 	var configOptions = this._translatorInfo.configOptions || {},
 		getCollections = configOptions.getCollections || false;
-	if(this._collection) {
-		this._itemGetter.setCollection(this._collection, getCollections);
-		delete this._collection;
-	} else if(this._items) {
-		this._itemGetter.setItems(this._items);
-		delete this._items;
-	} else {
-		this._itemGetter.setAll(getCollections);
+	switch ((this._export || {}).type) {
```

Comment:
It should be an error to call this without setting `this._export` first (may have to fix that in current code). Likewise, it should be an error to call this with unknown `this._export.type`, so the `default` should throw.